street_network/src/transform/classify_intersections.rs

```diff
@@ -0,0 +1,46 @@
+use crate::osm::NodeID;
+use crate::IntersectionComplexity::*;
+use crate::{IntersectionComplexity, OriginalRoad, RawIntersection, StreetNetwork};
+
+/// Determines the initial complexity of all intersections. Intersections marked "Crossing" are
+/// considered "unclassified" and will be updated with a guess, others will be left unchanged.
+pub fn classify_intersections(streets: &mut StreetNetwork) {
+    let mut changes: Vec<(NodeID, IntersectionComplexity)> = Vec::new();
+    for (id, inter) in streets.intersections.iter() {
+        if let Crossing = inter.complexity {
+            changes.push((
+                *id,
+                guess_complexity(inter, streets.roads_per_intersection(*id)),
+            ));
+        }
+    }
+
+    for (id, complexity) in changes.into_iter() {
+        if let Some(inter) = streets.intersections.get_mut(&id) {
```

Comment:
```suggestion
        streets.intersections.get_mut(&id).unwrap().complexity = complexity;
```
Slightly more succinct, and the unwrap is definitely safe because no IDs suddenly disappeared

Reply:
And our borrow of `streets` guarantees that! I had it that way first, and changed it because i forgot that the borrow checked has my back here.